### DISCUSSION TITLE
Custom Columns for Noticed::Event and Noticed::Notification

### DISCUSSION THREAD

**Discussion by t2** [excid3/noticed#517] [Q&A] [answered]
I am using the ActsAsTenant gem and would like to add the `account_id` to both the Noticed::Event and Noticed::Notification. I have the columns in the DB and the appropriate extensions created as shown here:

```ruby
module NoticedNotificationExtensions
  extend ActiveSupport::Concern

  included do
    acts_as_tenant(:account)
  end
end

module NoticedEventExtensions
  extend ActiveSupport::Concern

  included do
    acts_as_tenant(:account)
  end
end

Rails.application.config.to_prepare do
  Noticed::Event.include NoticedEventExtensions
  Noticed::Notification.include NoticedNotificationExtensions
end
```

However, when Sidekiq is sending the notification and doesn't have the `current_tenant` context, I would have assumed I could pass in the account as a param but I am getting a validation error. 

```ruby
Groups::NewMessageNotifier.with(
  message:,
  mentioned:,
  record: group,
  account: group.account
).deliver(member)
```

I know I can wrap it in `ActsAsTenant.with_tenant` but wanted to make sure there wasn't a simpler way before making those changes. Has anyone done this successfully?

**Comment by excid3** [comment, marked as answer]
`with` assigns attributes using `new` in ActiveRecord but it assigns them to `params`.

https://github.com/excid3/noticed/blob/5dc5842f223c8c1c5024c7ac5a52b5de40201060/app/models/concerns/noticed/deliverable.rb#L66-L73

I imagine this will work:

```ruby
notifier = Groups::NewMessageNotifier.with(
  message:,
  mentioned:,
  record: group,
  account: group.account
)
notifier.account = group.account
notifier.deliver(member)
```

Also just using `new` directly:

```ruby
Groups::NewMessageNotifier.new(
  params: {
    message:,
    mentioned:,
  },
  record: group,
  account: group.account
).deliver(member)
```

Sidekiq has an ActsAsTenant middleware that should set the current tenant though?

**Reply by t2** [reply]
You're right, the middleware should handle it... Not sure why it threw an error there. I think I need to do some more investigating. Thanks for the help!
